src/react-apps/applications/service-development/src/App.tsx

```diff
@@ -1,38 +1,127 @@
+  );
+};
+
 export interface IAppProps {
   drawerOpen: boolean;
 }
 
-const styles: any = {
-  root: {
-    background: '#EFEFEF',
-    top: '69px',
-  },
-  mainContent: {
-    paddingLeft: '72px',
-  },
-};
-
 class AppClass extends React.Component<IAppProps, any> {
 
   public handleDrawerToggle = () => {
     NavigationActionDispatcher.toggleDrawer();
   }
 
   public render() {
+
+    const altinnWindow: IAltinnWindow = window as IAltinnWindow;
+    const { org, service } = altinnWindow;
+    // const { org, service, instanceId, reportee } = altinnWindow;
+
+    const routes = [
+      {
+        path: '/om',
+        exact: true,
+        activeSubHeaderSelection: 'om',
+        menu: 'om',
+        subapp: DummySubApp,
+      },
+      {
+        path: '/uieditor',
+        activeSubHeaderSelection: 'lage',
+        menu: 'lage',
+        subapp: SubApp,
+      },
+      {
+        path: '/preview',
+        activeSubHeaderSelection: 'lage',
+        menu: 'lage',
+        subapp: SubApp,
+      },
+      {
+        path: '/sprak',
+        activeSubHeaderSelection: 'sprak',
+        menu: 'sprak',
+        subapp: DummySubApp,
+      },
+      {
+        path: '/teste',
+        activeSubHeaderSelection: 'teste',
+        menu: 'teste',
+        subapp: DummySubApp,
+      },
+      {
+        path: '/publisere',
+        activeSubHeaderSelection: 'publisere',
+        menu: 'publisere',
+        subapp: DummySubApp,
+      },
+    ];
+
     return (
-      <div style={styles.root}>
-        <NavMenu handleToggleDrawer={this.handleDrawerToggle} drawerOpen={this.props.drawerOpen} />
-        <div style={styles.mainContent}>
-          <SubApp />
-        </div>
-      </div>
+      <React.Fragment>
+        <Router>
+          <Grid container={true} direction='row'>
+            <Grid item={true} xs={12}>
+              <Route
+                exact={true}
+                path='/'
+                render={() => (
+                  <Redirect to='/om' />
+                )}
+              />
+              {routes.map((route, index) => (
+                <Route
+                  key={index}
+                  path={route.path}
+                  exact={route.exact}
+                  render={(props) => <AppBarComponent
+                    {...props}
+                    org={org}
+                    service={service}
+                    showSubHeader={true}
+                    activeSubHeaderSelection={route.activeSubHeaderSelection}
+                  />}
+                />
+              ))}
+            </Grid>
+            <Grid item={true} xs={12}>
+              <div style={{ display: 'flex', width: '100%', alignItems: 'stretch' }}>
+                <NavMenu handleToggleDrawer={this.handleDrawerToggle} drawerOpen={this.props.drawerOpen} />
+                <div style={{ paddingLeft: 10 }}>
+                  {routes.map((route, index) => (
+                    <Route
+                      key={index}
```

Comment:
Doesn't this throw an error? Since there is multiple elements with the same key? Or would that only happen inside the same element? (Since you map over these values 2 times, maybe move the mapping to a const?)

Reply:
No error.
Same key, different html tree, different route.